sycl/test-e2e/ESIMD/mask_compress_store.cpp

```diff
@@ -0,0 +1,86 @@
+//==-mask_compress_store.cpp - Test to verify compressed store functionality==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// RUN: %{build} -fsycl-device-code-split=per_kernel -o %t.out
+// RUN: %{run} %t.out
+
+// This is a basic test to validate the compressed store API.
+
+#include "esimd_test_utils.hpp"
+
+using namespace sycl;
+using namespace sycl::ext::intel::esimd;
+
+template <typename DataT>
+using shared_allocator = sycl::usm_allocator<DataT, sycl::usm::alloc::shared>;
+template <typename DataT>
+using shared_vector = std::vector<DataT, shared_allocator<DataT>>;
+
+template <int N> bool test(sycl::queue &Queue) {
+  shared_allocator<uint32_t> Allocator(Queue);
+
+  shared_vector<uint32_t> Output(N, 0, Allocator);
+  shared_vector<uint32_t> OutputVector(N, 0, Allocator);
+  std::vector<uint32_t> ExpectedOutput(Output.begin(), Output.end());
+  std::vector<uint32_t> OutputAcc(Output.begin(), Output.end());
+  int idx = 0;
+  for (int I = 0; I < N; I++) {
+    if ((I % 2) == 0)
+      ExpectedOutput[idx++] = I + 1;
+  }
+
+  auto *OutputPtr = Output.data();
+  auto *OutputVectorPtr = OutputVector.data();
+  buffer<uint32_t, 1> OutputAcc_buffer(OutputAcc.data(), OutputAcc.size());
+
+  Queue.submit([&](sycl::handler &cgh) {
+    auto OutputAcc_out = OutputAcc_buffer.get_access<access::mode::write>(cgh);
+    auto Kernel = ([=]() SYCL_ESIMD_KERNEL {
+      simd<uint32_t, N> Input(1, 1);
+      simd<uint32_t, N> Result(0);
+      simd_mask<N> Mask;
+      for (int i = 0; i < N; i++)
+        Mask[i] = (i % 2) == 0;
+      mask_compress_store(OutputPtr, Input, Mask);
+      mask_compress_store(Result, 0, Input, Mask);
+      mask_compress_store(OutputAcc_out, 0, Input, Mask);
+      Result.copy_to(OutputVectorPtr);
+    });
+    cgh.single_task(Kernel);
+  });
+  Queue.wait();
+
+  for (int I = 0; I < N; I++) {
+    if (Output[I] != OutputVector[I] && OutputAcc[I] != Output[I] &&
```

Comment:
Do we need to change the logic from `&&` to `||` to handle the case where the acc api works but the usm one fails?